PrismaDB-QueryAST/DML/Expressions/BooleanExpressions.cs

```diff
@@ -66,78 +73,246 @@ public override void setValue(params object[] value)
+                        search_index++;
+                        match = CaseInsensitiveCompare(svalue_store[search_index], col_store[column_index]);
+                        search_index++;
+                        column_index++;
+                        break;
+
+                    case EvalState.Percent:
+                        search_index = search_index + NextNonPercentIndex(svalue_store.Substring(search_index));
+                        if (search_index >= svalue_store.Length) return !NOT; //Trailing characters of search value consist only of %
+                        int search_chars = NextPercentIndex(svalue_store.Substring(search_index), esc_store);
+                        if (search_chars == -1)
+                        {
+                            if (CompareTrailingCharacters(svalue_store.Substring(search_index), col_store.Substring(column_index), esc_store)) return !NOT;
+                            else return NOT;
+                        }
+                        int matchingindex = FindNextMatch(svalue_store.Substring(search_index, search_chars), col_store.Substring(column_index), esc_store);
+                        if (matchingindex == -1) return NOT;  //Substring of search value does not exist in column value
+                        search_index = search_index + search_chars;
+                        column_index = column_index + matchingindex + 1;
+                        break;
+
+                    case EvalState.Underscore:
+                        search_index++;
+                        column_index++;
+                        break;
                 }
-                //Match intermediate characters
-                else
+                if (!match) return NOT;
+            }
+            return NOT;
+        }
+ 
+        private enum EvalState
+        {
+            Start,
+            Character,
+            Escape,
+            Percent,
+            Underscore,
+            Last,
+            End
+        }
+
+        private Boolean CaseInsensitiveCompare(char a, char b)
+        {
+            return Char.ToUpper(a) == Char.ToUpper(b);
+        }
+
+        private int NextPercentIndex(String search, String escape)
+        {
+            for (int i = 0; i < search.Length; i++)
+            {
+                if (search[i] == '%')
                 {
-                    containsindex = ContainsUnderscore(col_store, svalue_section);
-                    if (containsindex == -1) return NOT;
-                    col_store = col_store.Substring(containsindex + svalue_section.Length);
+                    Boolean escaped = false;
+                    for (int j = 0; j < escape.Length; j++)
+                    {
+                        if (search[i - 1] == escape[j]) escaped = true;
+                    }
+                    if (!escaped) return i;
                 }
+            }
+            return -1;
+        }
+
+        private int ConstraintedIndex(int search_index, int search_length)
```

Comment:
Constrained? :)

Reply:
oh LOL. Anyway Ill rename it to SetIndex or SetSearchIndex, i think it makes more sense too.